homeassistant/components/__init__.py

```diff
@@ -122,18 +142,40 @@ def handle_turn_service(service):
         yield from asyncio.wait(tasks, loop=hass.loop)
 
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_OFF, handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service)
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_ON, handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service)
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TOGGLE, handle_turn_service)
+        ha.DOMAIN, SERVICE_TOGGLE, async_handle_turn_service)
 
     @asyncio.coroutine
-    def handle_reload_config(call):
-        """Service handler for reloading core config."""
-        from homeassistant.exceptions import HomeAssistantError
-        from homeassistant import config as conf_util
+    def async_handle_core_service(call):
+        """Service handler for handling core services."""
+        if call.service == SERVICE_HOMEASSISTANT_STOP:
+            hass.exit_code = 0
+            hass.loop.create_task(hass.async_stop())
+            return
+
+        try:
+            yield from conf_util.async_check_ha_config_file(
+                hass, "{0}.{1}".format(ha.DOMAIN, call.service))
```

Comment:
This means there can be 2 notifications Icheck, restart) meaning the same thing.
I think we should use the same ID.